File: SeongYeon/baekjoon/1874.js
const fs = require('fs');
const inputFile = '/dev/stdin';
const input = fs
  .readFileSync(inputFile)
  .toString()
  .split('\n')
  .map(function (a) {
    return +a;
  });
const n = input[0];

const stack = [];
let number = 1;
let result = [];

for (let i = 1; i <= n; i++) {
  while (number <= input[i]) {
    stack.push(number);
    result.push('+');
    number++;
  }
  if (stack.pop() !== input[i]) {
    result = ['NO'];
    break;
  }
  result.push('-');
}

console.log(result.join('\n'));

File: JiHwan/BJ/1874.py
size = int(input())

stack = []
num = 0
answer = []
TF = "YES"

for i in range(size):
    x = int(input())

    while num < x:
        num += 1
        stack.append(num)
        answer.append('+')
        print(answer)
    
    if stack[-1] == x:
        stack.pop()
        answer.append('-')
        print(answer)
    
    else :
        TF = "NO"
        print(TF)
        break

if TF != "NO":
    for i in answer:
        print(i) 
    

